### DISCUSSION TITLE
What does `---@class PluginLspOpts` mean in `Lua`, or in the context of `lazyvim`?

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#1256] [Q&A]
As highlighted below, 

1. What does this type of syntax `---@` mean in `Lua`, such as `---@class PluginLspOpts`, or `---@type lspconfig.options`, or `---@param opts PluginLspOpts`? 

2. What do they mean in the context of `lazyvim`?

Thank you. 

<img width="1024" alt="Screenshot 2023-07-30 at 3 20 28 PM" src="https://github.com/LazyVim/LazyVim/assets/23435810/dbc63903-d033-4c72-a93a-ea4ab6d2ae8e">

**Comment by LintaoAmons** [comment]
You can find an overview of the lua_ls annotations here https://github.com/LuaLS/lua-language-server/wiki/Annotations
